src/EventSubscriber/InitSubscriber.php

```diff
@@ -138,7 +131,7 @@ public function checkRequiredPasswordChange(GetResponseEvent $event): void {
    */
   public function passwordChangeNotification(GetResponseEvent $event): void {
     if ($event->isMasterRequest() && Url::fromRoute('password_enhancements.password_change')->toString() !== $event->getRequest()->getPathInfo()) {
-      $policy = $this->policyStorage->loadByRoleAndPriority($this->account->getRoles());
+      $policy = PasswordPolicy::loadByRoles($this->account->getRoles());
       if ($policy !== NULL && $this->passwordChecker->showWarningMessage($policy)) {
         /** @var \Drupal\user\UserInterface $user */
         $user = $this->userStorage->load($this->account->id());
```

Comment:
What makes sure that in the next line `new FormattableMarkup() `does not get called with a `NULL` value or an empty string? Because `$policy->getExpiryWarningMessage()` currently can return both and displaying a message with "no message" does not make too much sense. As I can see `$this->passwordChecker->showWarningMessage($policy)` does not check whether `$policy->getExpiryWarningMessage()` returns a non-empty value.

Reply:
`password_enhancements_form_user_role_form_alter_validate()`